From 180e908d482635524b873c08e94658d22bf92cad Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Tue, 5 May 2015 09:55:30 +0800
Subject: [PATCH 1035/1594] MLK-10813-2: ASoC: imx-wm8958: fix issue that
 clock is not disabled

commit 7c3ca923f52956a3c279b9fab1ea90dc88d1445a from
git://git.freescale.com/imx/linux-2.6-imx.git

The clock disable has no chance to be called in set_bias_level_post,
because the dapm->bias_level is set by the codec driver, machine driver
should use the card->dapm.bias_level.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
---
 sound/soc/fsl/imx-wm8958.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/sound/soc/fsl/imx-wm8958.c b/sound/soc/fsl/imx-wm8958.c
index 443b7a9..1092444 100644
--- a/sound/soc/fsl/imx-wm8958.c
+++ b/sound/soc/fsl/imx-wm8958.c
@@ -352,7 +352,7 @@ static int imx_wm8958_set_bias_level(struct snd_soc_card *card,
 
 	switch (level) {
 	case SND_SOC_BIAS_STANDBY:
-		if (dapm->bias_level == SND_SOC_BIAS_OFF) {
+		if (card->dapm.bias_level == SND_SOC_BIAS_OFF) {
 			if (!IS_ERR(data->mclk)) {
 				ret = clk_prepare_enable(data->mclk);
 				if (ret) {
@@ -381,7 +381,7 @@ static int imx_wm8958_set_bias_level_post(struct snd_soc_card *card,
 
 	switch (level) {
 	case SND_SOC_BIAS_OFF:
-		if (dapm->bias_level == SND_SOC_BIAS_STANDBY)
+		if (card->dapm.bias_level == SND_SOC_BIAS_STANDBY)
 			if (!IS_ERR(data->mclk))
 				clk_disable_unprepare(data->mclk);
 		break;
@@ -389,7 +389,7 @@ static int imx_wm8958_set_bias_level_post(struct snd_soc_card *card,
 		break;
 	}
 
-	dapm->bias_level = level;
+	card->dapm.bias_level = level;
 
 	return 0;
 }
-- 
1.7.5.4

